File: server/Models/Cryptid.cs
using System.ComponentModel.DataAnnotations;

namespace cryptipedia.Models;

public class Cryptid : RepoItem<int> // int is the type of id
{
  // id, createdAt, updatedAt inherited from repo item
  public string Name { get; set; }
  [MaxLength(10000)] public string Description { get; set; }
  [Url, MaxLength(3000)] public string ImgUrl { get; set; }
  [Range(0, 10)] public int ThreatLevel { get; set; }
  [Range(0, 10)] public int Size { get; set; }
  public string Origin { get; set; }
  public string DiscovererId { get; set; }
  public Profile Discoverer { get; set; }
}

File: client/src/models/Cryptid.js
import { Profile } from "./Profile.js"

export class Cryptid {
  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.description = data.description
    this.imgUrl = data.imgUrl
    this.threatLevel = data.threatLevel
    this.origin = data.origin
    this.size = data.size
    this.discovererId = data.discovererId
    this.discoverer = new Profile(data.discoverer)
  }

  get displayId() {
    return this.id < 10 ? '0' + this.id : this.id
  }
}